src/iocore/net/ProxyProtocol.cc

```diff
@@ -291,7 +291,7 @@ proxy_protocol_v2_parse(ProxyProtocol *pp_info, const swoc::TextView &msg)
     }
 
     if (tlv_len > 0) {
-      if (pp_info->set_additional_data(msg.substr(msg.length() - tlv_len)) < 0) {
+      if (pp_info->set_additional_data(msg.substr(total_len - tlv_len, tlv_len)) < 0) {
```

Comment:
Are we sure that total_len > tlv_len?

Reply:
I think these lines ensure that the calculated value is always valid.
```
const size_t total_len = PPv2_CONNECTION_HEADER_LEN + len;

if (msg.size() < total_len) {
    return 0;
}

tlv_len = len - PPv2_ADDR_LEN_INET;
```